vaadin-grid-flow/src/main/resources/META-INF/resources/frontend/gridConnector.js

```diff
@@ -1,1017 +1,1023 @@
+      });
+
+      const _updateGridCache = tryCatchWrapper(function(page, items, callback, levelcache) {
+        // Force update unless there's a callback waiting
+        if (!callback) {
+          let rangeStart = page * grid.pageSize;
+          let rangeEnd = rangeStart + grid.pageSize;
+          if (!items) {
+            if (levelcache && levelcache.items) {
+              for (let idx = rangeStart; idx < rangeEnd; idx++) {
+                delete levelcache.items[idx];
+              }
+            }
+          } else {
+            if (levelcache && levelcache.items) {
+              for (let idx = rangeStart; idx < rangeEnd; idx++) {
+                if (levelcache.items[idx]) {
+                  levelcache.items[idx] = items[idx - rangeStart];
+                }
               }
             }
           }
         }
-      }
-    };
-
-    /**
-     * Updates all visible grid rows in DOM.
-     */
-    const updateAllGridRowsInDomBasedOnCache = function () {
-      grid._cache.updateSize();
-      grid._effectiveSize = grid._cache.effectiveSize;
-      grid._assignModels();
-    }
-
-    /**
-     * Update the given items in DOM if currently visible.
-     *
-     * @param array items the items to update in DOM
-     */
-    const updateGridItemsInDomBasedOnCache = function(items) {
-      if (!items || !grid._physicalItems) {
-        return;
-      }
+      });
+
       /**
-       * Calls the _assignModels function from GridScrollerElement, that triggers
-       * the internal revalidation of the items based on the _cache of the DataProviderMixin.
-       * First mapping the item to physical (iron list) indexes, so that we update
-       * only items in with the correct index that are cached in the iron list.
+       * Updates all visible grid rows in DOM.
        */
-      const itemKeys = items.map(item => item.key);
-      const indexes = grid._physicalItems
-          .map((tr, index) => tr._item && tr._item.key && itemKeys.indexOf(tr._item.key) > -1 ? index : null)
-          .filter(idx => idx !== null);
-      if (indexes.length > 0) {
-        grid._assignModels(indexes);
-      }
-    };
+      const updateAllGridRowsInDomBasedOnCache = tryCatchWrapper(function () {
+        grid._cache.updateSize();
+        grid._effectiveSize = grid._cache.effectiveSize;
+        grid._assignModels();
+      })
 
-    grid.$connector.set = function(index, items, parentKey) {
-      if (index % grid.pageSize != 0) {
-        throw 'Got new data to index ' + index + ' which is not aligned with the page size of ' + grid.pageSize;
-      }
-      let pkey = parentKey || root;
-
-      const firstPage = index / grid.pageSize;
-      const updatedPageCount = Math.ceil(items.length / grid.pageSize);
+      /**
+       * Update the given items in DOM if currently visible.
+       *
+       * @param array items the items to update in DOM
+       */
+      const updateGridItemsInDomBasedOnCache = tryCatchWrapper(function(items) {
```

Comment:
This seems to only be called from within other methods that are already using `tryCatchWrapper` so it could be removed here.

Reply:
Updated.